src/compute.rs

```diff
@@ -131,3 +125,41 @@ pub fn poly(x: &Scalar, f: &Vec<Point>) -> Result<Point, PointError> {
 
     Point::multimult(s, f.clone())
 }
+
+/// Create a BIP340 compliant tagged hash by double hashing the tag
+pub fn tagged_hash(tag: &str) -> Sha256 {
+    let mut hasher = Sha256::new();
+    let mut tag_hasher = Sha256::new();
+
+    tag_hasher.update(tag.as_bytes());
+    let tag_hash = tag_hasher.finalize();
+
+    hasher.update(tag_hash);
+    hasher.update(tag_hash);
+
+    hasher
+}
+
+/// Create a BIP341 compliant taproot tweak from a public key and merkle root
```

Comment:
I saw you changed this, but isn't BIP340 (Schnorr signatures) the one that defines the details of the tweak - while it's used in BIP341 (Taproot)?

Reply:
Exactly, BIP340 defines Schnorr, while BIP341 defines taproot, and BIP342 defines tapscript.

So `compute::{challenge, tagged_hash}` are BIP340, while `compute::{tweak, tweaked_public_key, merkle_root}` are BIP341.